triqsPackages/cppdlr/default.nix

```
@@ -0,0 +1,50 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, gtest
+, triqsPackages
+, fmt
+, python3Packages
+, openblasCompat
+, hdf5
+, mpi
+, git
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cppdlr";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "flatironinstitute";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-PxZtJifJotQtON+bEVGelXl/G9jxxfV5WmD6UJlhtLU=";
+  };
+
+  patches = [ ./cppdlr.patch ];
+  cmakeFlags = [ "-DBuild_Deps=Never" ];
+  nativeBuildInputs = [ cmake gtest git ];
+  buildInputs = [
+    hdf5
+    mpi
+    openblasCompat
+    fmt
+  ];
+
+  propagatedBuildInputs = [
+    triqsPackages.cpp2py
+    triqsPackages.h5
+    triqsPackages.nda
+    triqsPackages.itertools
+    triqsPackages.mpi
+  ];
```

Does cppdlr really depend on all of these? As far as I can see it only depends on `nda` and `h5`.
```suggestion
  propagatedBuildInputs = [
    triqsPackages.h5
    triqsPackages.nda
  ];
```

---

I think i wrongly interpreted the error message of a failing build of cppdlr complaining about some missing dependencies. That might however be due to these being dependencies of the dependencies that are not properly propagated. I will have a look into that

---

the installation of cppdlr automatically installs nda and therefore expects its dependencies (itertools, cpp2py and mpi) to be available. I have moved them from buildInputs to propagatedBuildInputs in the nda package or are they rather suited for propagatedNativeBuildInputs?